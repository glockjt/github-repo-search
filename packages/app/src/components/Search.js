import { useState } from "react";
import PropTypes from "prop-types";
import styled from "@emotion/styled";
/** @jsx jsx */
import { css, jsx } from "@emotion/core";

const Input = styled.input`
  padding: 16px 8px;
  font-size: 16px;
  width: 100%;
  border: 1px solid #d9e2ec;
  background-color: #d9e2ec;

  :focus {
    outline: none;
    border: 1px solid #2cb1bc;
    background-color: #fff;
  }
`;

const Button = styled.button`
  padding: 16px 32px;
  font-size: 16px;
  border: 1px solid #2cb1bc;
  background-color: #2cb1bc;
  color: #044e54;
  font-weight: 600;
`;

const Search = props => {
  const onClick = async () => {
    props.onClick(search);
    window.scrollTo(0, 0);
  };

  const onKeyDown = e => {
    if (e.key === "Enter") {
      onClick();
    }
  };

  const onChange = e => setSearch(e.target.value);

  const [search, setSearch] = useState("");
  return (
    <div
      css={css`
        display: flex;
        width: 100%;
        position: fixed;
        top: 0;
      `}
    >
      <Input
        placeholder="Search for repos"
        value={search}
        onKeyDown={onKeyDown}
        onChange={onChange}
      />
      <Button onClick={onClick}>Search</Button>
    </div>
  );
};

Search.propTypes = {
  onClick: PropTypes.func.isRequired
};

export default Search;
