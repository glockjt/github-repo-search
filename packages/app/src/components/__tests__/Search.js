import React from "react";
import renderer from "react-test-renderer";
import Search from "../Search";

describe("Search tests", () => {
  it("should render an empty search", () => {
    const tree = renderer.create(<Search onClick={jest.fn()} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
