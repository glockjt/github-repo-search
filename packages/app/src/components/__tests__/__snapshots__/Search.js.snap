// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search tests should render an empty search 1`] = `
<div
  className="css-g6h4xk-Search"
>
  <input
    className="css-1s785y2"
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder="Search for repos"
    value=""
  />
  <button
    className="css-1kq9qcs"
    onClick={[Function]}
  >
    Search
  </button>
</div>
`;
